### DISCUSSION TITLE
Group feature ?

### DISCUSSION THREAD

**Discussion by dgillier** [munafio/chatify#222] [General]
Hi,
This is a really great component !

Any idea if a group feature is planned, if yes when ?

Thanks, Denis

**Comment by munafio** [comment]
@dgillier Hi,👋🏻
Currently not planned to work on this feature .. It was planned to be released with Chatify v2, but because of the time and effort I'm spending on this project without any support, I stopped the plan just for now.
I will write an announcement very soon about the plan and this project, so stay tuned until then.

Have a good day.

**Reply by juliaokataleko** [reply]
@munafio how to contribute on this project? I need this feature! I can try to implement!

**Comment by dgillier** [comment]
Thanks @munafio for the clarification !

**Comment by hadad-ali** [comment]
Any news ?

**Comment by Mlaazizli** [comment]
we can help, I'm a Laravel fullstack developper

**Reply by munafio** [reply]
I appreciate any help from anyone, but for me actually not planning to add this feature in the near future because of the time.

**Comment by hadad-ali** [comment]
just tell me what to do , and I will send you the new code to check

Le jeu. 26 janv. 2023 à 20:39, Munaf A. Mahdi ***@***.***> a
écrit :

**Reply by munafio** [reply]
you can make a PR then I'll review the code

**Comment by munafio** [comment]
Annocement, 
Chatify was and will stay as one-to-one complete realtime chat system, and what I want to say that there is no plan to add a group feature in the future.
It was just a feature idea to add but been canceled. But you can implement a group/rooms feature yourself with Chatify since it is fully customizable and gives you a freedom to do what ever you want.

Regards, 
Munaf.

**Reply by Gizmo440** [reply]
Hello,
I can pay you for your service. How much would you charge to make me the source files for implementing a group/rooms feature so I can add it to my code? Thank you so much!

**Comment by rifrocket** [comment]
I was already started working on the group chat but due to so much other work load I was not able to finish that, and if it is not going to be merged with this repository I should work on another way! Maybe an add-on will do the work, I will start working on that as soon as I will get some time.

**Reply by dgillier** [reply]
That would be amazing, this is really the missing piece ;)

**Comment by skills-up** [comment]
@munafio if I were to add a group feature, what database structure would you propose?

I was thinking of three tables: groups (id, name, description, timestamps, meta), group_users (group_id, user_id, role_id/role_name), and group_messages (id, group_id, user_id, body, attachment, timestamps). I would use a roles/permissions library with this, or role my own.

Also, thereafter should it be a pub/sub or something else would be more appropriate in your opinion given the push based system?

**Comment by holmesadam** [comment]
If this had a group feature, the plugin would absolutely blow up - it's the missing feature that everyone needs. Like others have mentioned, happy to contribute towards this to get it over the line :)

**Reply by munafio** [reply]
You and everyone are welcome to contribute to this project.. if someone do it the right way, maybe will reconsider of rooms support.

**Reply by holmesadam** [reply]
Hi. You mention "the right way", can you provide some guidance on what you'd expect. I'd hate to spend a chunk of time on this to head down the wrong path. Cheers, Adam.

**Reply by munafio** [reply]
I meant to be implemented by following package's code, not making unnecessary changes, making it with a good UI/UX design.. this way I guess, because it was implemented before but has lots of issues and did not merge it.

**Comment by munafio** [comment]
We did a re-write for the package, now its way better and comes with the very wanted feature by you `Groups`! enjoy..

https://www.chatifyphp.com/
